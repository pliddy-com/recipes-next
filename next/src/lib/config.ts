interface PageProps {
  defaultDescription?: string;
  defaultTitle?: string;
  description?: string;
  domain?: string;
  title?: string;
}

interface MicrocopyProps {
  notFound: PageProps;
  index: PageProps;
  recipe: PageProps;
  site: PageProps;
  tag: PageProps;
  tagIndex: PageProps;
}

export interface ConfigProps {
  microcopy?: MicrocopyProps;
}

const microcopy: MicrocopyProps = {
  index: {
    defaultTitle: 'All the Recipes',
    description: "Patrick's recipe collection"
  },
  notFound: {
    defaultTitle: "Sorry, that page isn't here.",
    description: 'The requested url was not found'
  },
  recipe: {
    defaultDescription: 'A recipe for',
    defaultTitle: 'Recipe'
  },
  site: {
    title: "Patrick's Recipes",
    domain: 'https://recipes.pliddy.com'
  },
  tag: {
    defaultDescription: 'A collection of recipies tagged as',
    defaultTitle: 'Collection'
  },
  tagIndex: {
    defaultTitle: 'Tag Collections',
    description: 'All tag collections that contain recipes.'
  }
};

const config: ConfigProps = {
  microcopy
};

export default config;
