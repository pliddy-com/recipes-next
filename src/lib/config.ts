interface PageProps {
  defaultDescription?: string;
  defaultTitle?: string;
  description?: string;
  domain?: string;
  title?: string;
}

interface MicrocopyProps {
  category?: PageProps;
  index?: PageProps;
  recipe?: PageProps;
  site?: PageProps;
  tag?: PageProps;
}

export interface ConfigProps {
  microcopy?: MicrocopyProps;
}

const microcopy: MicrocopyProps = {
  category: {
    defaultDescription: 'A collection of recipies categorized as',
    defaultTitle: 'Category',
  },
  index: {
    defaultTitle: 'All Recipes',
    description: "Patrick's recipe collection",
  },
  recipe: {
    defaultDescription: 'A recipe for',
    defaultTitle: 'Recipe',
  },
  site: {
    title: "Patrick's Recipes",
    domain: 'https://recipes.pliddy.com',
  },
  tag: {
    defaultDescription: 'A collection of recipies tagged as',
    defaultTitle: 'Tag',
  },
};

const config: ConfigProps = {
  microcopy,
};

export default config;
