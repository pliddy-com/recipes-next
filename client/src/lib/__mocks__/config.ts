import { ConfigProps } from 'lib/config';

const microcopy = {
  index: {
    defaultTitle: 'Index Default Title',
    description: 'Index default description.'
  },
  notFound: {
    defaultTitle: 'Page Not Found',
    description: 'The requested url was not found'
  },
  recipe: {
    defaultDescription: 'Recipe default description for',
    defaultTitle: 'Recipe Default Title'
  },
  search: {
    defaultTitle: 'Results for',
    description: 'Search results'
  },
  site: {
    title: 'Site Title',
    domain: 'https://recipes.pliddy.com'
  },
  tag: {
    defaultDescription: 'Tag default description for',
    defaultTitle: 'Tag Default Title'
  },
  tagIndex: {
    defaultTitle: 'Tag Index Default Title',
    description: 'Tag index default description.'
  }
};

const config: ConfigProps = {
  microcopy
};

module.exports = config;
