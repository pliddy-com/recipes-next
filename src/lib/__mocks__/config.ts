import { ConfigProps } from 'lib/config';

const microcopy = {
  category: {
    defaultDescription: 'Category default description',
    defaultTitle: 'Category Default Title',
  },
  index: {
    defaultTitle: 'Index Default Title',
    description: 'Index default description.',
  },
  recipe: {
    defaultDescription: 'Recipe default description',
    defaultTitle: 'Recipe Default Title',
  },
  site: {
    title: 'Site Title',
  },
  tag: {
    defaultDescription: 'Tag default description',
    defaultTitle: 'Tag Default Title',
  },
};

const config: ConfigProps = {
  microcopy,
};

module.exports = config;
