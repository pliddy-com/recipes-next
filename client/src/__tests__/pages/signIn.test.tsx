import '@testing-library/jest-dom';
import { act, render, waitFor } from '@testing-library/react';
import preloadAll from 'jest-next-dynamic';

import SignInPage from 'pages/signin';
import config from 'lib/config';

jest.mock('lib/config');
jest.mock('components/SignInForm/SignInForm');

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    push: jest.fn()
  })
}));

describe('SignInPage in signIn.tsx', () => {
  beforeEach(async () => {
    await preloadAll();
  });

  afterEach(() => {
    jest.resetModules();
  });

  describe('when there is content', () => {
    it('it renders the signin page', async () => {
      const { asFragment, queryByTestId } = render(
        // <AuthenticationProvider>
        <SignInPage />
        // </AuthenticationProvider>
      );

      // wait for dynamic component to load
      await act(async () => waitFor(() => queryByTestId('page')));

      //   expect(routerSpy).toBeCalled();

      // wait for dynamic component to load
      await act(async () => waitFor(() => queryByTestId('signInForm')));

      // assert that the component matches the existing snapshot
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when there is no config object', () => {
    // before each test, delete microcopy node from config
    beforeEach(() => {
      delete config.microcopy;
    });

    it('it does not render the page', async () => {
      const { queryByTestId } = render(<SignInPage />);

      // wait for dynamic component to load
      await act(async () =>
        waitFor(() => expect(queryByTestId('page')).toBeNull())
      );
    });
  });
});
