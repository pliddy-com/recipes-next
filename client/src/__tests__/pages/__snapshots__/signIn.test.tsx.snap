// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInPage in signIn.tsx when there is content it renders the signin page 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl page recipegrid css-19r6kue-MuiContainer-root"
    data-testid="page"
  >
    <title>
      Site Title | Sign In
    </title>
    <meta
      content="Site sign in"
      name="description"
    />
    <meta
      content="Site sign in"
      property="og:description"
    />
    <meta
      content="en_US"
      property="og:locale"
    />
    <meta
      content="Site Title"
      property="og:site_name"
    />
    <meta
      content="Sign In"
      property="og:title"
    />
    <meta
      content="website"
      property="og:type"
    />
    <meta
      content="https://131-auth-integration.recipes.pliddy.comundefined"
      property="og:url"
    />
    <link
      href="https://131-auth-integration.recipes.pliddy.comundefined"
      rel="canonical"
    />
    <h1
      class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
    >
      Sign In
    </h1>
    <div
      class="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
    />
  </div>
</DocumentFragment>
`;
